### PR TITLE
Support for Signing using .NET Core 2.1+

### PR DESCRIPTION
**Pull request by clairernovotny**
cc @dtivel 

This PR has the changes necessary to enable signing using .NET Core 2.1 and higher. 

Mostly for reference when you're ready.

### CODE REVIEW

**Comment by glennawatson** [comment]
The embedded is a nice touch for the pdbtype for making debugging a bit easier.

File: src/NuGet.Core/NuGet.Packaging/Signing/Authoring/X509SignatureProvider.cs [hunk 1]
```diff
@@ -141,7 +141,7 @@ private static PrimarySignature CreatePrimarySignature(CmsSigner cmsSigner, Sign
 
             try
             {
-                cms.ComputeSignature(cmsSigner);
+                cms.ComputeSignature(cmsSigner, false); // silent is false to ensure PIN prompts appear if CNG/CAPI requires it
```
Comment (clairernovotny) [inline src/NuGet.Core/NuGet.Packaging/Signing/Authoring/X509SignatureProvider.cs:144]: @PatoBeltran as you're working on .net core 2.1 signing stuff, please don't forget this change too.

**Comment by zkat** [comment]
Hey @onovotny! Thanks so much for your work on this PR and your interest in getting this working.

That said, this needs a bit more than just updating the conditionals and adjusting a couple of things. We've found that a lot of tests need to be updated (and in many cases, the associated functionality fixed). There's currently a large team effort going to enable cross-platform signing, and we expect it to become available by sometime around netcore5 preview 2, or so.

You can check out our toplevel epic for this work over at https://github.com/NuGet/Home/issues/7939 (you will need to have ZenHub installed in order to see the epic issues associated with this). You can also see the work in the `dev-feature-xplatVerification` feature branch in this repo.

Since this work is being done separately, I'd like to thank you again for the work you've done, and I'm going to close this particular PR in favor of the ongoing team work. Have a good one!

File: src/NuGet.Core/NuGet.Packaging/Signing/Authoring/X509SignatureProvider.cs (hunk 1 above)
Comment (zkat) [reply src/NuGet.Core/NuGet.Packaging/Signing/Authoring/X509SignatureProvider.cs:144]: p.s. we'll make sure to replicate this change! Can you tell me more about why this is needed? I'm still fairly new to this work and the codebase, myself :)

we want to make sure we also add appropriate tests for this.

Comment (clairernovotny) [reply src/NuGet.Core/NuGet.Packaging/Signing/Authoring/X509SignatureProvider.cs:144]: The issue is that if you use a smart card/fob that requires an interactive prompt for a password, you need to make sure it appears.

I'm not quite sure how to automate a test for it, but in my case, I have a certificate on a fob that uses the SafeNet Authentication Client. When a cert operation is invoked, it prompts a dialog for the password to the certificate.

Without my change, the silent is true, meaning it won't prompt and the operation will fail.

Comment (nkolev92) [reply src/NuGet.Core/NuGet.Packaging/Signing/Authoring/X509SignatureProvider.cs:144]: I think @onovotny has an equivalent PR for the feature branch. 

https://github.com/NuGet/NuGet.Client/pull/3069

Comment (glennawatson) [reply src/NuGet.Core/NuGet.Packaging/Signing/Authoring/X509SignatureProvider.cs:144]: Anyone who uses yubikey for their certificate will need the false flag set as well.

**Pull request closed without merging**